src/product/tests/product.controller.spec.ts

```diff
@@ -0,0 +1,93 @@
+import { Test } from '@nestjs/testing';
+
+import { ProductController } from '../product.controller';
+import { ProductService } from '../product.service';
+import { CreateProductDto } from '../dto/create-product.dto';
+import { PaginationDto } from '../../common/dto/pagination.dto';
+import { Sorting } from '../../common/types/pagination';
+import { ProductState } from '../types/product';
+import { UpdateProductDto } from '../dto';
+
+describe('ProductController', () => {
+  let controller: ProductController;
+  let service: ProductService;
+
+  beforeEach(async () => {
+    const mockService = {
+      findOne: jest.fn(),
+      findAll: jest.fn(),
+      create: jest.fn(),
+      update: jest.fn(),
+      remove: jest.fn(),
+    };
+    const module = await Test.createTestingModule({
+      controllers: [ProductController],
+      providers: [
+        {
+          provide: ProductService,
+          useValue: mockService,
+        },
+      ],
+    }).compile();
+
+    controller = module.get<ProductController>(ProductController);
+    service = module.get<ProductService>(ProductService);
+  });
+
+  it('should be defined', () => {
+    expect(controller).toBeDefined();
+  });
+
+  describe('findOne', () => {
+    it('should call productService.findOne with correct params', async () => {
+      const searchTerm = 'someTerm';
+      await controller.findOne(searchTerm);
+      expect(service.findOne).toHaveBeenCalledWith(searchTerm);
+    });
+  });
+
+  describe('findAll', () => {
+    it('should call productService.findAll with correct params', async () => {
+      const paginationDto: PaginationDto = {
+        limit: 10,
+        offset: 0,
+        sort: Sorting.ASC,
+      };
+      await controller.findAll(paginationDto);
+      expect(service.findAll).toHaveBeenCalledWith(paginationDto);
+    });
+  });
+
+  describe('create', () => {
+    it('should call productService.create with correct params', async () => {
+      const createProductDto: CreateProductDto = {
```

Comment:
The dtos should also be moved to a separate mock file.
Also, WDYT about using faker for the mocks data?

Reply:
good idea

Reply:
done